controllers/k8ssandra/secrets.go

```diff
@@ -60,9 +63,12 @@ func (r *K8ssandraClusterReconciler) reconcileReaperSecrets(ctx context.Context,
 				uiUserSecretRef.Name = reaper.DefaultUiSecretName(kc.SanitizedName())
 			}
 			kcKey := utils.GetKey(kc)
-			if err := secret.ReconcileSecret(ctx, r.Client, cassandraUserSecretRef.Name, kcKey); err != nil {
-				logger.Error(err, "Failed to reconcile Reaper CQL user secret", "ReaperCassandraUserSecretRef", cassandraUserSecretRef)
-				return result.Error(err)
+			if kc.Spec.Reaper != nil && kc.Spec.Reaper.ReaperRef.Name == "" {
```

Comment:
@adejanovski , I had to revert this back to the ReaperRef check.
If I checked only a storage type, then a bundled reaper that uses local storage was not working properly.